### PR TITLE
[CI] Move gha artifact download before xml parsing for test stat uploads

### PR DESCRIPTION
**Pull request by clee2000**
Move gha artifact download to before any xml parsing is done for uplaod-test-stats

Do not download gha artifacts during xml parsing since got uploaded to s3 in the above and will be downloaded when all the artifacts are downloaded from s3

The previous method resulted in dups if you run the script again  

TODO: write a deduper so we don't have to worry at all

### CODE REVIEW

File: tools/stats/upload_test_stats.py [hunk 1]
```diff
@@ -123,12 +122,6 @@ def get_tests(workflow_run_id: int, workflow_run_attempt: int) -> List[Dict[str,
         for path in s3_paths:
             unzip(path)
 
-        artifact_paths = download_gha_artifacts(
```
Comment (huydhn) [inline tools/stats/upload_test_stats.py:126]: We probably have them all on S3 now with ODIC, but plz double check edge cases like ROCm

**Review by huydhn — APPROVED** [review]
LGTM!

File: tools/stats/upload_test_stats.py (hunk 1 above)
Comment (clee2000) [reply tools/stats/upload_test_stats.py:126]: I'm confused, I thought all the artifacts would eventually get uploaded to s3 by the step that uploads from gha artifacts to s3?

Comment (huydhn) [reply tools/stats/upload_test_stats.py:126]: Yup, that's my thought too.  Although the workflow needs to be added there explicitly https://github.com/pytorch/pytorch/blob/main/.github/workflows/upload-test-stats.yml#L5

**Pull request closed without merging**